docs/template_plugin/tests/functional/op_reference/random_uniform.cpp

```diff
@@ -0,0 +1,212 @@
+    ReferenceRandomUniformLayerTest,
+    ::testing::Values(
+        RandomUniformParams(
+            std::vector<int64_t>{3, 2, 4},
+            Tensor{{1}, element::f32, std::vector<float>{0}},
+            Tensor{{1}, element::f32, std::vector<float>{1}},
+            element::Type_t::f32,
+            150,
+            10,
+            Tensor{{3, 2, 4},
+                   element::f32,
+                   std::vector<float>{
+                       0.7011235952377319336, 0.3053963184356689453, 0.9393105506896972656, 0.9456034898757934570,
+                       0.1169477701187133789, 0.5077005624771118164, 0.5197197198867797852, 0.2272746562957763672,
+                       0.9913740158081054688, 0.3551903963088989258, 0.8269231319427490234, 0.5986485481262207031,
+                       0.3136410713195800781, 0.5748131275177001953, 0.4139908552169799805, 0.9630825519561767578,
+                       0.3714079856872558594, 0.8525316715240478516, 0.0935858488082885742, 0.0820095539093017578,
+                       0.2365508079528808594, 0.8105630874633789062, 0.7422660589218139648, 0.7610669136047363281}},
+            "float32_default_min_max"),
+        RandomUniformParams(std::vector<int64_t>{3, 2, 4},
+                            Tensor{{1}, element::f16, std::vector<float16>{0}},
+                            Tensor{{1}, element::f16, std::vector<float16>{1}},
+                            element::Type_t::f16,
+                            150,
+                            10,
+                            Tensor{{3, 2, 4},
+                                   element::f16,
+                                   std::vector<float16>{0.6044921875, 0.8066406250, 0.8320312500, 0.3837890625,
+                                                        0.0361328125, 0.0830078125, 0.5439453125, 0.8339843750,
+                                                        0.3359375000, 0.7197265625, 0.1542968750, 0.1289062500,
+                                                        0.3476562500, 0.8691406250, 0.4130859375, 0.5722656250,
+                                                        0.5742187500, 0.9394531250, 0.6552734375, 0.8222656250,
+                                                        0.8242187500, 0.1328125000, 0.6435546875, 0.6601562500}},
+                            "float16_default_min_max"),
+        RandomUniformParams(std::vector<int64_t>{3, 2, 4},
+                            Tensor{{1}, element::f32, std::vector<float>{-650}},
+                            Tensor{{1}, element::f32, std::vector<float>{450}},
+                            element::Type_t::f32,
+                            150,
+                            10,
+                            Tensor{{3, 2, 4},
+                                   element::f32,
+                                   std::vector<float>{
+                                       121.2359619140625000000,  -314.0640563964843750000, 383.2415771484375000000,
```

Comment:
Do we really need to have so much numbers after the dot? If I remember right the float type has less precision